pkg/kademlia/kademlia.go

```diff
@@ -222,60 +186,58 @@ func (k *Kademlia) Ping(ctx context.Context, node pb.Node) (pb.Node, error) {
 	if err != nil {
 		return pb.Node{}, NodeErr.Wrap(err)
 	}
-
 	if !ok {
 		return pb.Node{}, NodeErr.New("Failed pinging node")
 	}
-
 	return node, nil
 }
 
 // FindNode looks up the provided NodeID first in the local Node, and if it is not found
 // begins searching the network for the NodeID. Returns and error if node was not found
 func (k *Kademlia) FindNode(ctx context.Context, ID storj.NodeID) (pb.Node, error) {
+	return k.lookup(ctx, ID, false)
+}
+
+//lookup initiates a kadmelia node lookup
+func (k *Kademlia) lookup(ctx context.Context, ID storj.NodeID, isBootstrap bool) (pb.Node, error) {
 	kb := k.routingTable.K()
-	nodes, err := k.routingTable.FindNear(ID, kb)
-	if err != nil {
-		return pb.Node{}, err
+	var nodes []*pb.Node
+	if isBootstrap {
+		for _, v := range k.bootstrapNodes {
+			nodes = append(nodes, &v)
+		}
+	} else {
+		var err error
+		nodes, err = k.routingTable.FindNear(ID, kb)
+		if err != nil {
+			return pb.Node{}, err
+		}
 	}
-
 	lookup := newPeerDiscovery(k.log, nodes, k.nodeClient, ID, discoveryOptions{
-		concurrency: k.alpha, retries: defaultRetries, bootstrap: false, bootstrapNodes: k.bootstrapNodes,
+		concurrency: k.alpha, retries: defaultRetries, bootstrap: isBootstrap, bootstrapNodes: k.bootstrapNodes,
 	})
-
 	target, err := lookup.Run(ctx)
 	if err != nil {
 		return pb.Node{}, err
 	}
+	bucket, err := k.routingTable.getKBucketID(ID)
+	if err != nil {
+		k.log.Warn("Error getting getKBucketID in kad lookup")
```

Comment:
why is this error not returned? (1)

Reply:
I guess the question is, what benefit would we get by propagating it up?  It's never supposed to happen, but the worst case effect is that we do extra work ever hour.  We definitely wouldn't want things to stop working because of this.  But we would want to know about it.  DPanic() might be a better choice?

Reply:
No it's fine, and that makes sense. I was just wondering.